tests/src/test/java/stepDefinitions/backendstepdefinitions/GenerateEfilingUrl.java

```diff
@@ -99,22 +102,43 @@ public static void user_calls_incorrect_with_POST_http_request(String resource)
         APIResources resourceInvalid = APIResources.valueOf(resource);
 
         request = given().spec(TestUtil.requestSpecification()).body(payloadData.generateUrlFinalPayload());
-        response = request.when().post(resourceInvalid.getResource() + "s").then().extract().response();
+        response = request.when().post(resourceInvalid.getResource()).then().extract().response();
     }
 
     @When("status is {int} and content type is verified")
     public static void status_is_and_content_type_is_verified(Integer int1) {
-        assertEquals(response.getStatusCode(), 404);
-        assertEquals(response.getContentType(), "application/json");
+        if(int1 == 404) {
+            assertEquals(response.getStatusCode(), 404);
+            assertEquals(response.getContentType(), CONTENT_TYPE);
+        } else if(int1 == 405) {
+            assertEquals(response.getStatusCode(), 405);
+            assertEquals(response.getContentType(), CONTENT_TYPE);
+        }
     }
 
-    @Then("verify response payload objects are correct")
-    public static void verify_response_payload_objects_are_correct() {
+    @Then("verify error message is present and message has no value")
+    public static void verify_error_message_is_present_and_message_has_no_value() {
+        jsonPath = new JsonPath(response.asString());
+
         String error = TestUtil.getJsonPath(response, "error");
         String message = TestUtil.getJsonPath(response, "message");
+        int status = jsonPath.get("status");
+
+        if(status == 404) {
```

Comment:
prefer using a map over if else, otherwise that might grow a lot ;-)

Reply:
Missed this sorry. Thank you. Will use when there is more that two to compare.